src/io/image_reader.cpp

```diff
@@ -22,7 +23,7 @@ bool IsJpegFile(std::string_view path) {
     return path.substr(path.find_last_of(".") + 1) == "jpg" || path.substr(path.find_last_of(".") + 1) == "jpeg";
 }
 
-deepworks::Tensor ReadJpegFile(std::string_view path) {
+deepworks::Tensor ReadJpegFile(std::string_view path, std::optional<deepworks::Tensor> out_tensor) {
```

Comment:
The same: it can be simple decomposed like this:
```cpp

Tensor ReadImage(std::string_view path) {
    Tensor out; // empty tensor
    ReadJpegFile(path, out); // allocate inside according the size
    return out;
}

void ReadimageToTensor(std::string_view path, Tensor& out) {
    // in simple case user don't know the image size and pass the empty tensor.
    // If tensor not empty and size isn't suitable it must throw possibly
    // If size is suitable just write to this memory
    ReadJpegFile(path, out);
}

// Usage

Tensor X;
readImageToTensor("bird.png", X); // just allocate the tensor
readImageToTensor("dog.png", X); // rewrite the tensor in case images have the same size

// or 
X = readImage("bird.png");
readImageToTensor("dog.png", X):
```

Reply:
* `readImageToTensor` can be used for `MNIST` dataset because we know that images has `28x28` size.
* `readImage` can be used if we don't know about image size